mountpoint-s3/src/inode.rs

```diff
@@ -507,6 +520,31 @@ pub struct LookedUp {
     pub stat: InodeStat,
 }
 
+impl Ord for LookedUp {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        other.inode.name().cmp(self.inode.name())
+    }
+}
+
+impl PartialOrd for LookedUp {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Eq for LookedUp {}
+
+impl PartialEq for LookedUp {
+    fn eq(&self, other: &Self) -> bool {
+        self.inode.ino() == other.inode.ino()
+            && self.inode.parent() == other.inode.parent()
+            && self.inode.name() == other.inode.name()
+            && self.inode.full_key() == other.inode.full_key()
+            && self.inode.kind() == other.inode.kind()
+            && self.stat == other.stat
+    }
+}
```

Comment:
`PartialOrd` and `PartialEq` [need to be consistent with each other](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html), which these aren't. In particular, the first rule is that `a == b` if and only if `partial_cmp(a, b) == Some(Equal)`, which isn't true here.